File: OJ practices/LeetCode题目集/392. Is Subsequence(easy).cpp
/**
 * @file 392. Is Subsequence(easy).cpp
 * @author chadchang (chadchang@tencent.com)
 * @brief 
 * @version 1.0.0
 * @date 2024-03-28
 * 
 * @copyright Copyright (c) 2024 SourDumplings
 * 
 * https://leetcode.cn/problems/is-subsequence/description/?envType=study-plan-v2&envId=top-interview-150
 * 
 */

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int i = 0, j = 0;
        int ls = s.length(), lt = t.length();
        while (i < ls)
        {
            while (j < lt && s[i] != t[j])
            {
                ++j;
            }
            if (j == lt)
            {
                return false;
            }
            ++i;
            ++j;            
        }
        return true;
    }
};

File: OJ practices/LeetCode题目集/392. Is Subsequence(easy).java
/*
 * @Author: SourDumplings
 * @Date: 2019-09-21 17:42:07
 * @Link: https://github.com/SourDumplings/
 * @Email: changzheng300@foxmail.com
 * @Description: https://leetcode.com/problems/is-subsequence/
 */

class Solution
{
    public boolean isSubsequence(String s, String t)
    {
        int j = 0;
        int ls = s.length();
        int lt = t.length();
        if (lt == 0 && ls > 0)
        {
            return false;
        }
        for (int i = 0; i < ls; i++)
        {
            char ci = s.charAt(i);
            for (; j < lt; j++)
            {
                char cj = t.charAt(j);
                if (ci == cj)
                {
                    ++j;
                    break;
                }
            }
            if (j == lt && ci != t.charAt(j - 1))
            {
                return false;
            }
        }
        return true;
    }
}